joomla/joomla-cms
# Statistics plugin improvements

## Description

This changes the JFormField to:
- Add a new `render()` function that can be used to render layouts inside other layouts. 
- Allow to easily override the layout include paths in any child field creating an override of the  `getLayoutPaths()` method.

Render layouts inside other layouts is used like:

``` php
<a href="http://joomla.org">Joomla</a>
<?php 
    // Render `sample.layout` layout inside this layout
    echo $field->render('sample.layout', $displayData);
?>
<p>Another paragraph</p>
```

You can see below how I used it to render a common `stats` layout that is renderer in both `message` and `field.data` layouts.

This PR also improves the statistics system plugin:
- Changed plugin fields prefix to plgsystemstats to avoid conflicts with any extension using `stats` prefix. Before: `type="stats.uniqueid"` after: `type="plgsystemstats.uniqueid"`
- Remove uniqueid field getInput() method and use the fields layout system so 100% of the plugin output is overridable.
- Create new field data that will be used to render the stats that are going to be sent in the plugin settings view.

![stats-params-data](https://cloud.githubusercontent.com/assets/1119272/11106927/4e93abc4-88d9-11e5-90cf-1b5ad6e6ed89.png)
## How to test
1. This changes the field layout system so you should test that fields that are currently using layouts work as expected. The only fields using it are `JFormFieldRadio` & `JFormFieldCheckboxes` so the best way to check is to edit the codemirror plugin that includes both fields in its settings.
2. Enable debug in statistics system plugin and test that the message selecting the user to configure statistics mode is shown. Also confirm that when you click in `Click here to see which information will be sent.` data is shown below.
3. Go to statistics system plugin settings and confirm that when you click in `Click here to see which information will be sent.` data appears below.
